src/Core/ExecutablePluginTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesPluginTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait ExecutablePluginTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getLabelValue() {
+    $definition = $this->getPluginDefinition();
+    if (empty($definition['label'])) {
+      throw new InvalidPluginDefinitionException('The label is not defined.');
```

Comment:
This is quite a useless exception. The label of which plugin ID? Which plugin type? Please include that as parameters in the message so that developers know where the exception comes from.